src/handlers/applicationFormsHandlers.ts

```diff
@@ -22,27 +22,103 @@ import type {
   ApplicationForm,
   ApplicationFormWrite,
   ApplicationFormField,
+  ApplicationFormWithFields,
 } from '../types';
 
 const logger = getLogger(__filename);
 
+export const getFieldFromRow = (row: ApplicationFormWithFields): ApplicationFormField => {
+  if (row.applicationFormFieldId === undefined
+    || row.applicationFormFieldId === null
+    || row.applicationFormFieldCanonicalFieldId === undefined
+    || row.applicationFormFieldCanonicalFieldId === null
+    || row.applicationFormFieldPosition === undefined
+    || row.applicationFormFieldPosition === null
+    || row.applicationFormFieldLabel === undefined
+    || row.applicationFormFieldLabel === null
+    || row.applicationFormFieldCreatedAt === undefined
+    || row.applicationFormFieldCreatedAt === null) {
+    throw new Error('Expected id and formField values to be present.');
+  }
+  return {
+    id: row.applicationFormFieldId,
+    applicationFormId: row.id,
+    canonicalFieldId: row.applicationFormFieldCanonicalFieldId,
+    position: row.applicationFormFieldPosition,
+    label: row.applicationFormFieldLabel,
+    createdAt: row.applicationFormFieldCreatedAt,
+  };
+};
+
 const getApplicationForms = (
   req: Request,
   res: Response,
   next: NextFunction,
 ): void => {
   db.sql('applicationForms.selectAll')
-    .then((applicationFormsQueryResult: Result<ApplicationForm>) => {
-      logger.debug(applicationFormsQueryResult);
-      const { rows: applicationForms } = applicationFormsQueryResult;
-      if (isApplicationFormArray(applicationForms)) {
+    .then((applicationFormsQueryResult: Result<ApplicationFormWithFields>) => {
+      const { rows: applicationFormsWithFields } = applicationFormsQueryResult;
+      if (isApplicationFormWithFieldsArray(applicationFormsWithFields)) {
+        let currentApplicationForm: ApplicationForm = {
+          id: 0,
+          opportunityId: 0,
+          version: -1,
+          createdAt: new Date(),
+          fields: [],
+        };
+        const applicationForms: ApplicationForm[] = [];
+        // First, read the ApplicationForm part from this row and compare to
+        // previous row (simply by the id). If it's the same, add only a field.
+        // If it's new, create a new ApplicationForm and add the first field if present.
+        applicationFormsWithFields.forEach((formWithFieldRow) => {
+          if (currentApplicationForm.id === formWithFieldRow.id) {
+            if (formWithFieldRow.applicationFormFieldId !== undefined
+              && formWithFieldRow.applicationFormFieldId !== null) {
+              const formField = getFieldFromRow(formWithFieldRow);
+              // add field to previousRow
+              if (currentApplicationForm.fields === undefined) {
+                currentApplicationForm.fields = [formField];
+              } else {
```

Comment:
Suppose I remove the above unreachable code (because it is always defined, you can see it defined on lines 67 and 96 here) in both places, now the compiler gets confused because it thinks it could possibly be undefined (even though it is always defined):

> src/handlers/applicationFormsHandlers.ts:79:15 - error TS2532: Object is possibly 'undefined'.
>
> 79               currentApplicationForm.fields.push(formField);
>                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>
> src/handlers/applicationFormsHandlers.ts:98:15 - error TS2532: Object is possibly 'undefined'.
>
> 98               currentApplicationForm.fields.push(formField);
>                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~